src/native/external/zlib-ng-version.txt

```diff
@@ -1,12 +1,14 @@
-v2.1.6
-74253725f884e2424a0dd8ae3f69896d5377f325
+v2.2.1
+d54e3769be0c522015b784eca2af258b1c026107
 
-https://github.com/zlib-ng/zlib-ng/releases/tag/2.1.6
+https://github.com/zlib-ng/zlib-ng/releases/tag/2.2.1
 
 We have removed the following folders from our local copy as these files are not needed for our compilation:
 
-- zlib-ng/docs/
+- zlib-ng/doc/
 - zlib-ng/test/
 - zlib-ng/arch/s390/self-hosted-builder/
 
-Apply https://github.com/dotnet/runtime/pull/105433.patch
\ No newline at end of file
+Also apply:
+- https://github.com/dotnet/runtime/commit/ecdb625035e0e3fb7c51e908713d96d2cb2080c8.patch or cherry-pick ecdb625035e0e3fb7c51e908713d96d2cb2080c8 directly.
+- https://github.com/dotnet/runtime/pull/105771.patch
```

Comment:
This links to a full 1000's lines bug patch. I think this should only link to a single commit with the specific change, similar to the previous line.

Reply:
Hold on, I'll have to squash everything. Sigh.

Reply:
That should do it.

Reply:
I am sorry, this is still the wrong 1000's lines commit.

 These commits should be only the zlib-ng delta vs. upstream.

Reply:
No worries, I want to do the right thing, but I am unsure how to get the correct delta that you expect.

The commit I used for the patch is the commit containing the squashed contents of this PR. That patch is showing a diff before and after updating zlib-ng (which does have a lot of changes since their last release). It's also excluding all the unnecessary files and folders, and also updating our informational json and txt files.

What am I missing?

Note: This comment thread is already old and pointing at the wrong patch (it's the patch of the PR itself). Are you looking at my most recent two commits?

Reply:
Yes, I am looking at the two most recent commits. 

Take a look at https://github.com/dotnet/runtime/pull/102231 for a good way to make these types of updates. Could you please update this PR to be composed of these 3 commits:

Commit 1: Verbatim copy of the zlib-ng v2.2.1 sources (with the specific directories and files excluded)
Commit 2: Our patches in zlib-ng sources. You can either keep the patches as multiple commits or they can be squashed into a single commit. Either way is fine.
Commit 3: Changes outside src/native/external/zlib-ng, including links to commit(s) 2 in zlib-ng-version.txt

It is nice to merge these PRs as "merge" instead of "squash" so that it is easier to tell what happened.

Reply:
Ok, done. If we merge this with a "merge-commit", the patch comment in zlib-ng-version.txt will be usable on the next zlib-ng update.